src/spider/worker/ExecutorHandle.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef SPIDER_WORKER_EXECUTORHANDLE_HPP
+#define SPIDER_WORKER_EXECUTORHANDLE_HPP
+
+#include <mutex>
+#include <optional>
+
+#include <boost/uuid/uuid.hpp>
+
+#include "TaskExecutor.hpp"
+
+namespace spider::worker {
+/**
+ * This class acts as a handle for thread-safe access to the task executor and task id.
+ * It maintains a weak reference to the task executor to prevent multiple destructor calls and
+ * ensures that access remains valid only while the executor itself is valid.
+ */
+class ExecutorHandle {
+public:
+    [[nodiscard]] auto get_task_id() -> std::optional<boost::uuids::uuid>;
+    [[nodiscard]] auto get_executor() -> TaskExecutor*;
+    auto executor_cancel() -> void;
+    auto set(boost::uuids::uuid task_id, TaskExecutor* executor) -> void;
+    auto clear() -> void;
+
+private:
+    boost::uuids::uuid m_task_id;
+    TaskExecutor* m_executor
+            = nullptr;  // Do not use std::shared_ptr to avoid calling destructor twice.
+    std::mutex m_mutex;
```

Comment:
With this design it seems easy to accidentally create multiple handles (and therefore mutexes) for the same TaskExecutor.

Is it possible to design things in a way where a handle has unique ownership of the TaskExecutor? It feels possible since currently it looks like each thread gets a reference to the handle.